plugins/imgur.py

```diff
@@ -128,7 +129,15 @@ def export_submission(self,
             if is_album:
                 results['link_display'] = '[Imgur Album](https://imgur.com/a/%s)  \n' % album['id']
             else:
-                results['link_display'] = '[Imgur](%s)  \n' % images[0]['link'].replace('http', 'https')
+                
+                picture_url = images[0]['link'].replace('http', 'https')
+                r = requests.head(picture_url)
+                mime_text = r.headers.get('Content-Type')
+                mime = mimeparse.parse_mime_type(mime_text)
+                if mime[1] == 'gif':
+                    picture_url = re.sub('\.gif|jpg|png', '.gifv', picture_url)
```

Comment:
Why not just do a `rsplit` and replace there? More importantly, though, `'\.` is an invalid escape sequence and will be deprecated. Either use a raw string or double-backslash.